aea/package_manager/v1.py

```diff
@@ -152,21 +170,39 @@ def sync(
 
     def update_package_hashes(self) -> "PackageManagerV1":
         """Update package.json file."""
-        for package_id, package_hash in self.get_available_package_hashes().items():
-            is_dev_package = package_id in self._dev_packages
-            is_third_party_package = package_id in self._third_party_packages
+
+        for package_id in self.iter_dependency_tree():
+            is_dev_package = self.is_dev_package(package_id=package_id)
+            is_third_party_package = self.is_third_party_package(package_id=package_id)
             if not is_dev_package and not is_third_party_package:
                 raise PackageNotValid(
                     f"Found a package which is not listed in the `packages.json` with package id {package_id}"
                 )
 
+            self.update_dependencies(
+                package_id=package_id,
+            )
+
+            package_hash = self.calculate_hash_from_package_id(package_id=package_id)
             if is_dev_package:
                 if self._dev_packages[package_id] == package_hash:
                     continue
 
                 self._logger.info(f"Updating hash for {package_id}")
                 self._dev_packages[package_id] = package_hash
 
+            if is_third_party_package:
+                if self._third_party_packages[package_id] == package_hash:
+                    continue
+
+                self._third_party_packages[package_id] = package_hash
+                self._logger.warning(
+                    "Hash change detected for third party package"
+                    f"\n\tPackage: {package_id}"
+                    f"\n\tCalculated hash: {package_hash}"
+                    f"\n\tExpected hash: {self._third_party_packages[package_id]}"
+                )
```

Comment:
@DavidMinarsch We used to ignore the third party hash changes which caused the issues you mentioned in the debugging channel. From now we will update the hashes with a warning so the developer is aware that the third party hash change was performed

Reply:
I don't quite follow. First, I am not sure your naming convention is correct. Do you actually mean "third party", i.e. a package from another author, or do you mean something else? Second, why did we not have this from the start? What was the rational to not include it. Third, why the warning? What is the rational to warn about it?

Reply:
Yes third party means packages from other authors which we don't intend to update

We had this because people were updating hashes for third party packages by mistake causing issues which were really hard to resolve so we prevented the hash update for the third party packages

The warning because of the said reason, we don't want to update the hashes for third party packages by mistake so showing a warning at least let's user know that there was a change in the third party package